### PR TITLE
RavenDB-19937: CompactTree throws `Invalid data found in the buffer` when trying to remove key.

### PR DESCRIPTION
**Pull request by redknightlois**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-19937

### Additional description
Decoding was done improperly as if it were an integer, therefore long values would fail. 

### Type of change
Bug fix

### How risky is the change?
Low 

### Backward compatibility
Non breaking change

### CODE REVIEW

File: src/Voron/Data/CompactTrees/CompactTree.cs [hunk 1]
```diff
@@ -1958,7 +1958,11 @@ int GetEncodedEntryUnlikely(byte* p, out int length)
 
             key = new Span<byte>(entryPos + lenOfKeyLen, keyLen);
             entryPos += keyLen + lenOfKeyLen;
-            value = ZigZagEncoding.Decode<long>(entryPos, out var valLen);
+
+            value = ZigZagEncoding.DecodeCompact<long>(entryPos, out var valLen, out var success);
```
Comment (ayende) [inline src/Voron/Data/CompactTrees/CompactTree.cs:1962]: Shouldn't this be the same behavior? What is the difference?

Comment (redknightlois) [reply src/Voron/Data/CompactTrees/CompactTree.cs:1962]: I was not checking for error explicitly.

Comment (ayende) [reply src/Voron/Data/CompactTrees/CompactTree.cs:1962]: How would you _get_ the error, however?

Comment (redknightlois) [reply src/Voron/Data/CompactTrees/CompactTree.cs:1962]: Different error, now we throw the same exception in both cases. Also notice that we are using the `Compact` version now (shorter codegen).

**Comment by arekpalinski** [comment]
Some tests are failing but I think they shouldn't be part of this PR (the ones about `null` support - https://github.com/ravendb/ravendb/pull/15770)

**Comment by redknightlois** [comment]
No, they are not supposed to.

**Comment by redknightlois** [comment]
Superseded by https://github.com/ravendb/ravendb/pull/15770

**Pull request closed without merging**